app/src/lib/git/stash.ts

```diff
@@ -1,7 +1,81 @@
+import { git } from '.'
+import { Repository } from '../../models/repository'
+
+export const DesktopStashEntryMarker = '!!GitHub_Desktop'
+
 export interface IStashEntry {
   /** The name of the branch at the time the entry was created. */
   readonly branchName: string
 
   /** The SHA of the commit object created as a result of stashing. */
   readonly stashSha: string
 }
+
+/** RegEx for parsing out the stash SHA and message */
+const stashEntryRe = /^([0-9a-f]{5,40})@(.+)$/
+
+/**
+ * RegEx for determining if a stash entry is created by Desktop
+ *
+ * This is done by looking for a magic string with the following
+ * format: `!!GitHub_Desktop<branch@commit>`
+ */
+const stashEntryMessageRe = /^!!GitHub_Desktop<(.+)@([0-9|a-z|A-Z]{5,40})>$/
```

Comment:
This format starts with `!!GitHub_Desktop` to reduce the chances of Desktop picking up stash entries created outside of Desktop. 

I expect the first part (left-hand side of `@` symbol) to be the canonical ref name and the second part to be what `HEAD` was pointing to at the time the entry was created.

While I will not be using the tip commit, I've included it in the format for a because it makes debugging issues around stashing easier since we will have a bit more info.

Reply:
:art: stylistic question that I can't not ask - why underscore here rather than hyphen?

Reply:
> `([0-9|a-z|A-Z]{5,40})`

Also, is there a situation where you expect the commit ID to not be 40 characters?

Reply:
> 🎨 stylistic question that I can't not ask - why underscore here rather than hyphen?

There is no reason 😆

Reply:
> > `([0-9|a-z|A-Z]{5,40})`
> 
> Also, is there a situation where you expect the commit ID to not be 40 characters?

Nope, will update to 40.